### PR TITLE
feat(ci): clean up test images after closing PRs

### PR DESCRIPTION
**Pull request by aali309**
# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
* [x] Read the [contributing guidelines](https://github.com/cryostatio/cryostat/blob/main/CONTRIBUTING.md)
* [x] Linked a relevant issue which this PR resolves
* [x] Linked any other relevant issues, PR's, or documentation, if any
* [x] Resolved all conflicts, if any
* [x] Rebased your branch PR on top of the latest upstream `main` branch
* [x] Attached at least one of the following labels to the PR: `[chore, ci, docs, feat, fix, test]`
* [x] [Signed all commits](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits): `git commit -S -m "YOUR_COMMIT_MESSAGE"`
_______________________________________________

Fixes: #603 

## Description of the change:
This change allows the removal of test images after the PR is closed

## Motivation for the change:
This change is helpful because it helps get rid of polluted dangling images

## How to manually test:
1. *Insert steps here...*
2. *...*

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -19,6 +19,7 @@ on:
       - synchronize
       - labeled
       - unlabeled
+      - closed
```
Comment (tthvo) [inline .github/workflows/ci.yaml:22]: we should split these changes into a workflow file.

Comment (tthvo) [reply .github/workflows/ci.yaml:22]: Like this:

https://github.com/cryostatio/cryostat/blob/main/.github/workflows/image-cleanup.yml

File: .github/workflows/image-cleanup.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  pull_request_target:
+    types:
+      - closed
+
+jobs:
+  delete-images:
+    name: Delete PR-scoped test images
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
```
Comment (tthvo) [inline .github/workflows/image-cleanup.yml:13]: We don't need this step as we just want to delete the images.

File: .github/workflows/image-cleanup.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        uses: actions/checkout@v2
+
+      - name: Delete Docker images
+        uses: r26d/ghcr-delete-image-action@v1.2.2
+        with:
+          owner: ${{ github.repository_owner }}
+          name: cryostat-operator
```
Comment (tthvo) [inline .github/workflows/image-cleanup.yml:20]: You need to delete a set of images, specifically:

- `cryostat-operator`
- `cryostat-operator-bundle`
- `cryostat-operator-scorecard`

You can try matrix:

https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs

File: .github/workflows/image-cleanup.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+          token: ${{ secrets.GHCR_PR_TOKEN }}
+          ignore-missing-package: true
+          tag-regex: pr-${{ github.event.number }}-.*
```
Comment (tthvo) [inline .github/workflows/image-cleanup.yml:23]: Good idea! Please update the test image tags in `ci.yaml` to include this prefix. For eaxmple, the bundle image:

https://github.com/cryostatio/cryostat-operator/blob/2d386930dc96f0dcaf937987ec35874006c53b61/.github/workflows/ci.yaml#L120

Comment (tthvo) [reply .github/workflows/image-cleanup.yml:23]: I would search for any tag with `ci-$GITHUB_SHA` and replace them with `pr-${{ github.event.number }}-$GITHUB_SHA`

**Review by tthvo — COMMENTED** [review]
Looks good! Once we address the new comments, can you provide a sample run?

File: .github/workflows/image-cleanup.yml [hunk 5]
```diff
@@ -9,20 +9,24 @@ jobs:
   delete-images:
     name: Delete PR-scoped test images
     runs-on: ubuntu-latest
-    steps:
-      - name: Check out code
-        uses: actions/checkout@v2
 
+    strategy:
+      matrix:
+       image-names:
```
Comment (tthvo) [inline .github/workflows/image-cleanup.yml:15]: Lets use it in array form and call it `matrix.image`

```yaml
       image: [cryostat-operator, cryostat-operator-bundle, cryostat-operator-scorecard]
````

Comment (aali309) [reply .github/workflows/image-cleanup.yml:15]: yah this looks cleaner

File: .github/workflows/image-cleanup.yml [hunk 6]
```diff
@@ -0,0 +1,25 @@
+name: Clean up PR-scoped test images
+
+on:
+  pull_request_target:
+    types:
+      - closed
+
+jobs:
+  delete-images:
+    name: Delete PR-scoped test images
```
Comment (tthvo) [inline .github/workflows/image-cleanup.yml:10]: Remove this since the workflow name already has it.

File: .github/workflows/image-cleanup.yml [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    runs-on: ubuntu-latest
+    if: github.repository_owner == 'cryostatio'
+    strategy:
+      matrix:
+        image: [cryostat-operator, cryostat-operator-bundle, cryostat-operator-scorecard]
+    steps:
+    - name: Delete Docker images
```
Comment (tthvo) [inline .github/workflows/image-cleanup.yml:17]: Remove this name too.

**Review by andrewazores — COMMENTED** [review]
Looks like it makes sense. Any sample run?

**Comment by aali309** [comment]
> Looks like it makes sense. Any sample run?

Yes, we had a meeting at 1pm, me and @tthvo. He showed me how I can run and make a PR on my fork. Made a few runs that failed and then it passed. I closed the PRs, see if I can retrieve them and the results.

**Review by andrewazores — APPROVED** [review]

**Comment by ebaron** [comment]
@aali309 great work, thanks for doing it!

**Pull request merged**